dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +147,233 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    async def get_task_ids_running_in_a_worker(self, worker_id: str) -> List[str]:
+        """
+        Retrieves the task IDs of running tasks associated with a specific worker.
+
+        Args:
+            worker_id (str): The ID of the worker.
+
+        Returns:
+            List[str]: A list containing the task IDs of all the running tasks associated with the worker.
+        """
+
+        option = ListApiOptions(
+            filters=[("worker_id", "=", worker_id), ("state", "=", "RUNNING")],
+            detail=True,
+            timeout=10,
+        )
+        ## Call the state API to get all tasks in a worker
+        tasks_in_a_worker_result = await self._state_api.list_tasks(option=option)
+        tasks_in_a_worker = tasks_in_a_worker_result.result
+
+        ## Get task_id from each task in a worker
+        task_ids_in_a_worker = []
+        if tasks_in_a_worker is not None:
```

Comment:
this won't be None right?

Reply:
You are right, even though all the tasks related to this working are not running, we would return an empty array from 
await self._state_api.list_tasks(option=option), which is safe

```
   option = ListApiOptions(
            filters=[("worker_id", "=", worker_id), ("state", "=", "RUNNING")],
            detail=True,
            timeout=10,
        )
        ## Call the state API to get all tasks in a worker
        tasks_in_a_worker_result = await self._state_api.list_tasks(option=option)
        tasks_in_a_worker = tasks_in_a_worker_result.result
```